lib/minidoc/finders.rb

```diff
@@ -24,17 +24,16 @@ def exists?(selector = {})
 
     def find(id_or_selector, options = {})
       if id_or_selector.is_a?(Hash)
-        options.merge!(transformer: method(:wrap))
-        collection.find(id_or_selector, options)
+        collection.find(id_or_selector, options).lazy.map(&method(:wrap))
```

Comment:
transformer was implemented in https://github.com/mongodb/mongo-ruby-driver/commit/53ad43fedc99efa15deff36e8a222f833f2b22f9 although I can't find exactly where it was removed

The idea with `lazy` is that we want to avoid reading all of the matching documents from the collection when this method is called - we want to leave that up to the consumer of this method. If we didn't have lazy, our `map` would cause all of the database queries to occur. The way the transformer worked, and the way this lazy map works (:fingers_crossed:), we get to defer wrapping the document in our model object until the last moment. That way things like `User.find(name: "Gordon").take(2)` should return objects of type `User` and minimize database queries.